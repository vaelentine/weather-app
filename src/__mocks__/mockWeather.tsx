const mockWeather =
  {
    coord: {
      lon: -122.6762,
      lat: 45.5234
    },
    weather: [
      {
        id: 800,
        main: 'Clear',
        description: 'clear sky',
        icon: '01n'
      }
    ],
    base: 'stations',
    main: {
      temp: 291.01,
      feels_like: 290.26,
      temp_min: 286.82,
      temp_max: 293.11,
      pressure: 1013,
      humidity: 54
    },
    visibility: 10000,
    wind: {
      speed: 0.89,
      deg: 313,
      gust: 4.02
    },
    clouds: {
      all: 1
    },
    dt: 1630560308,
    sys: {
      type: 2,
      id: 2008548,
      country: 'US',
      sunrise: 1630503161,
      sunset: 1630550935
    },
    timezone: -25200,
    id: 5746545,
    name: 'Portland',
    cod: 200
  }

export default mockWeather
