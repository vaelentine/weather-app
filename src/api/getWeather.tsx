import axios from 'axios'
import * as dotenv from "dotenv";
dotenv.config({ path: __dirname+'/.env' });
// require('dotenv').config()

async function getWeather(city: string) {
  try {
    const server = 'openweathermap.org/'
    const weatherEndpoint = 'data/2.5/weather'
    const keyParam = `appid=${process.env.REACT_APP_WEATHER_API_KEY}`
    const request = `https://api.${server}${weatherEndpoint}?${keyParam}&q=${city}&units=imperial`
    const response = await axios.get(request, {})
    return response
  } catch (error:any) {
    return error.response
  }
}

export default getWeather
