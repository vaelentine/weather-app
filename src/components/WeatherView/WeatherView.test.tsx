/* eslint-disable no-undef */
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Weatherview from './WeatherView'
import getWeather from 'api/getWeather'
import mockWeather from '__mocks__/mockWeather'

describe('WeatherView', () => {
  it('renders correctly', () => {
    const data = mockWeather
    const component = render(<Weatherview weatherData={ data }/>)
    expect(component.asFragment()).toMatchSnapshot()
  })
})
