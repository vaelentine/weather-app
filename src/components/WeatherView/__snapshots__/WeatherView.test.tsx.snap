// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherView renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weatherContainer"
  >
    <div
      class="weatherDisplay"
    >
      <div
        class="weatherHeader"
      >
        <div
          class="cityName"
        >
          Portland
        </div>
        <div
          class="timeStamp"
        >
          1630560308
        </div>
      </div>
      <div
        class="currentTemp"
      >
        291.01°F
      </div>
      <div
        class="minMaxTemps"
      >
        <div>
          Low: °F
        </div>
        <div>
          High: °F
        </div>
      </div>
      <div
        class="icon"
      >
        <div
          class="weatherIcon"
        >
          <img
            alt="Clear"
            class="iconImage"
            src="https://openweathermap.org/img/wn/01n.png"
          />
        </div>
      </div>
      <div
        class="weatherInfo"
      >
        <div
          class="weatherDescription"
        >
          clear sky
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
